src/compiler/checker.ts

```diff
@@ -7395,7 +7395,7 @@ namespace ts {
 
         function getConstrainedTypeVariable(typeVariable: TypeVariable, node: Node) {
             let constraints: Type[];
-            while (isPartOfTypeNode(node)) {
+            while (node && !isStatement(node)) {

```

Comment:
Is `isExpression` preferable to `isStatement` here?

Reply:
No, there might not be an enclosing expression anywhere in the parent chain.

Reply:
We could *additionally* check for `!isExpression(node)` but it isn't really necessary, we'll eventually hit a statement (which could be a declaration statement) or the very top node.